checker/manual/external-tools.tex

```diff
@@ -444,37 +444,106 @@ \section{Maven\label{maven}}
-\code{compile.options.fork.executable = "\$CHECKERFRAMEWORK/checker/bin/javac"}
+See the directory \code{examples/GradleExamples/} for examples of running a
+checker in a Gradle project.  You can use these examples to verify that
+Gradle is correctly downloading and executing the Checker Framework.
 
-To specify command-line arguments, set
-\code{compile.options.compilerArgs}.  Here is a possible example:
+Please note that the \<-AoutputArgsToFile> command-line option
+(see Section~\ref{debugging-options-output-args}) and shorthands for built-in checkers
+(see Section~\ref{shorthand-for-checkers}) are not available when
+following these instructions.  Both these features are available only when a checker is
+launched via \<checker.jar> such as when \code{\$CHECKERFRAMEWORK/checker/bin/javac}
+is run.  The Gradle instructions in this section
+bypass \<checker.jar> and cause the compiler to run a
+checker as an annotation processor directly.
+
+These instructions can be used with Java 7 or Java 8.  If using Java 7, then
+type annotations should be placed in comments as explained in
+Section~\ref{annotations-in-comments}.
+
+\begin{enumerate}
+
+\item Add a property to indicate which version of Java to which the source and
+class files are targeted.  This property can be set based on the current version
+of Java as shown below or hard code.
 
 \begin{Verbatim}
-allprojects {
-  tasks.withType(JavaCompile).all { JavaCompile compile ->
-    compile.options.debug = true
-    compile.options.compilerArgs = [
-      '-version',
-      '-implicit:class',
-      '-processor', 'org.checkerframework.checker.nullness.NullnessChecker'
-    ]
-    options.fork = true
-    options.forkOptions.executable = "$CHECKERFRAMEWORK/checker/bin/javac"
-  }
+ext.targetJavaVersion = JavaVersion.current().isJava7() ? JavaVersion.VERSION_1_7 : JavaVersion.VERSION_1_8
+\end{Verbatim}
+
+\item Define the Maven Central repostiory:
+
+\begin{Verbatim}
+repositories {
+  ... extisting repositories...
+  mavenCentral()
+}
+\end{Verbatim}
+
+\item Add dependency configurations for the annotated JDK, \<checker.jar>, and the Type Annotations compiler:
+
+\begin{Verbatim}
+configurations {
+    ... extisting configurations ...
+    if (targetJavaVersion.isJava7()) {
+        checkerFrameworkJavac {
+            description = 'a customization of the Open JDK javac compiler with additional support for type annotations'
+        }
+    }
+    checkerFrameworkAnnotatedJDK {
+       description = 'a copy of JDK classes with Checker Framework type qualifers inserted'
+    }
+    checkerFramework {
+       description = 'The Checker Framework: custom pluggable types for Java'
+    }
+}
+\end{Verbatim}
+
+\item Using the Maven coordinates, declare the Checker Framework dependencies:
+
+\begin{Verbatim}
+dependencies {
+    ... extisting dependencies...
+    ext.checkerFrameworkVersion = '1.9.11'
+    ext.jdkVersion = JavaVersion.current().isJava7() ? 'jdk7' : 'jdk8'
```

Comment:
Should this use `ext.targetJavaVersion` instead of `JavaVersion.current()`?  Same question in the example code.